config/routes.rb

```diff
@@ -3,4 +3,10 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  get '/users', to: 'users#index'  # For displaying all users
+  get '/users/:id', to: 'users#show', as: 'user'  # For displaying a single user
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts' # For displaying a user's postsget '/users/:user_id/posts/:post_id', to: 'posts#show', as: 'user_post'
+  get '/users/:user_id/posts/:post_id', to: 'posts#show', as: 'user_post' # For displaying a specific post by user
```

Comment:
- [ ] [OPTIONAL] Kindly write the routes in this way:
```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

as it's cleaner and easier to read and maintain.

Reply:
Thanks very much sir, will do.